enhancements/cluster-api/cluster-autoscaler-integration-with-openshift-cluster-api.md

```diff
@@ -0,0 +1,751 @@
+
+#### Security
+
+This change will allow users to specify MachineSet references that will exist in two
+namespaces. Although the entry is not free form, and the reference will be validated
+by the Cluster Autoscaler Operator, the OpenShift Cloud Infrastructure team has
+reviewed this design for possible security vulnerabilities and considers the risk low.
+
+Before releasing this change, it will be reviewed by product security for final analysis.
+
+#### User Experience
+
+This enhancement describes a small change in the user experience with most of the
+feature work landing in the Cluster Autoscaler Operator. The Cloud Infrastructure has
+designed this change to follow the current user experience without introducing
+new concepts for enabling autoscaling on any specific MachineSet resource.
+
+The addition of conditions to the status will improve the user experience by giving
+direct feedback on warning and errors that may arise during operation. The Cloud
+Infrastructure team has designed this change to reduce confusion for users and to
+provide a clear path for triage.
+
+Outside of the Cluster Autoscaler Operator, users will need to understand that the
+Cluster Autoscaler is now focused on Cluster API MachineSets instead of Machine API
+MachineSets. This change will require users to understand the authoritative and
+mirroring mechanisms around the Cluster API migration to maximize their operation
+and maintenance activities.
+
+These changes should be clear for users, but to help mitigate the risk of misinterpretation,
+the user experience focused changes in this enhancement will be documented in the
+official OpenShift documentation.
+
+### Drawbacks
+
+A drawback of the approach defined in this enhancement is that it could cause
+confusion among users. The change in API behavior introduced in this enhancement
+will allow users to specify the MachineSet resource of their choice when
+enabling autoscaling. Due to the mirrored nature of the Machine API and Cluster API
+MachineSets, users may not immediately comprehend which resource they should
+specify, and what workflow is recommended by Red Hat.
+
+To reduce confusion, the changes described in this enhancement will be reviewed in
+the OpenShift product documentation, with examples to clarfiy the intended usage.
+A post on the OpenShift developer blog (https://developers.redhat.com/blog) to
+describe this feature in greater depth could be added to any Cluster API change
+post threads to help add informational material.
+
+Another approach to reducing confusion would be to allow only a single type of
+MachineSet resource (Machine API or Cluster API) to be specified as a target for
+autoscaling. This approach could work if the Cluster API resources are chosen as
+the target, but would represent a hard shift in the current MachineAutoscaler
+behavior and would require a conversion migration for all upgrades where cluster
+autoscaler is in use.
```

Comment:
Would it? If the CAO and KAS always wrote to CAPI, that would be predictable at least?